api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -49,216 +55,627 @@ public StorageController(IOptions<StorageOptions> options)
+                    client.Connect();
+                    var safePath = Path.Combine(locationPath, path);
+                    if (!client.Exists(safePath)) throw new InvalidOperationException($"File does not exist: '{path}'");
+
+                    var info = new ItemModel(Path.GetFileName(safePath), client.GetAttributes(safePath));
+                    client.Delete(safePath);
+                    return new JsonResult(info);
+                }
+                finally
+                {
+                    if (client.IsConnected)
+                        client.Disconnect();
+                }
+            }
+            else if (connection == null || connection?.ConnectionType == ConnectionType.LocalVolume)
+            {
+                var safePath = Path.Combine(_options.GetCapturePath(), path.MakeRelativePath());
+                if (!safePath.FileExists() && !safePath.DirectoryExists()) throw new InvalidOperationException($"File/folder does not exist: '{path}'");
+
+                // TODO: Only certain users should be allowed to delete certain files/folders.
+                var item = new ItemModel(safePath);
+                if (item.IsDirectory) Directory.Delete(safePath);
+                else System.IO.File.Delete(safePath);
+                return new JsonResult(item);
+            }
+            else throw new NotImplementedException($"Location connection type '{connection?.ConnectionType}' not implemented yet.");
+        }
+        else
+        {
+            var safePath = Path.Combine(_options.GetUploadPath(), path.MakeRelativePath());
+            if (!safePath.FileExists() && !safePath.DirectoryExists()) throw new InvalidOperationException($"File/folder does not exist: '{path}'");
+
+            // TODO: Only certain users should be allowed to delete certain files/folders.
+            var item = new ItemModel(safePath);
+            if (item.IsDirectory) Directory.Delete(safePath);
+            else System.IO.File.Delete(safePath);
+            return new JsonResult(item);
+        }
     }
 
     /// <summary>
     /// Make a clip from the target file identified in the clip parameter 'prefix'.
     /// </summary>
+    /// <param name="locationId"></param>
     /// <param name="path"></param>
     /// <param name="start"></param>
     /// <param name="end"></param>
     /// <param name="outputName"></param>
     /// <returns></returns>
     [HttpPost("clip")]
+    [HttpPost("{locationId:int}/clip")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(ItemModel), (int)HttpStatusCode.OK)]
     [ProducesResponseType((int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Storage" })]
-    public async Task<IActionResult> CreateClipAsync(string path, int start, int end, string outputName)
+    public async Task<IActionResult> CreateClipAsync([FromRoute] int? locationId, [FromQuery] string path, [FromQuery] int start, [FromQuery] int end, [FromQuery] string outputName)
```

Comment:
I wish we could execute FFmpeg commands directly on the remote location, but it doesn't work.  Need to copy locally first.